### PR TITLE
Build and publish iOS framework in GitHub release files

### PR DESCRIPTION
**Pull request by reuben**

### CODE REVIEW

File: taskcluster/ios-package.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+mkdir -p ${TASKCLUSTER_ARTIFACTS} || true
+
+cp ${DS_ROOT_TASK}/DeepSpeech/ds/tensorflow/bazel*.log ${TASKCLUSTER_ARTIFACTS}/
+
+package_native_client "native_client.tar.xz"
+
+package_libdeepspeech_as_zip "libdeepspeech.zip"
+
+case $arch in
```
Comment (lissyx) [inline taskcluster/ios-package.sh:17]: nit: do we really need the `case` here? I feel like we could just have
```
  ${TAR} -cf - \
         -C ${DS_ROOT_TASK}/DeepSpeech/ds/native_client/swift/DerivedData/Build/Products/Release-*/ deepspeech_ios.framework \
         | ${XZ} > "${TASKCLUSTER_ARTIFACTS}/deepspeech_ios.framework.${arch}.tar.xz"
```

File: taskcluster/tc-build-utils.sh [hunk 2]
```diff
@@ -315,3 +315,18 @@ do_nuget_build()
 
   nuget pack nupkg/deepspeech.nuspec
 }
+
+do_deepspeech_ios_framework_build()
+{
+  arch=$1
+  cp ${DS_TFDIR}/bazel-bin/native_client/libdeepspeech.so ${DS_DSDIR}/native_client/swift/libdeepspeech.so
+  cd ${DS_DSDIR}/native_client/swift
+  case $arch in
```
Comment (lissyx) [inline taskcluster/tc-build-utils.sh:324]: nit: I think we could just have the case set a few variables like the SDK variant and have one `xcodebuild` call after:

```
  case $arch in
  "--x86_64")
    iosSdk="iphonesimulator"
    ;;
  "--arm64")
    iosSdk="iphoneos"
    ;;
  esac
xcodebuild -workspace deepspeech_ios.xcworkspace -scheme deepspeech_ios_test -configuration Release -arch ${arch} -sdk ${iosSdk} -derivedDataPath DerivedData CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO CODE_SIGNING_ALLOWED=NO
```

File: taskcluster/ios-package.sh [hunk 3]
```diff
@@ -16,13 +16,13 @@ package_libdeepspeech_as_zip "libdeepspeech.zip"
...
 "--arm64")
-  ${TAR} -cf - \
-         -C ${DS_ROOT_TASK}/DeepSpeech/ds/native_client/swift/DerivedData/Build/Products/Release-iphoneos/ deepspeech_ios.framework \
-         | ${XZ} > "${TASKCLUSTER_ARTIFACTS}/deepspeech_ios.framework.arm64.tar.xz"
+  artifact_name="deepspeech_ios.framework.arm64.tar.xz"
 ;;
 esac
+
+${TAR} -cf - \
+       -C ${DS_ROOT_TASK}/DeepSpeech/ds/native_client/swift/DerivedData/Build/Products/Release-iphonesimulator/ deepspeech_ios.framework \
```
Comment (lissyx) [inline taskcluster/ios-package.sh:27]: Aren't you going to have an issue here with `Release-iphonesimulator` ? ARM64 builds seems to end up in `Release-iphoneos` ?

Comment (reuben) [reply taskcluster/ios-package.sh:27]: ugh. nice catch

**Pull request merged**